tests/everflow/everflow_test_utilities.py

```diff
@@ -303,10 +301,96 @@ def acl_stage(self):
         Get the ACL stage for this set of test cases.
 
         Used to parametrize test cases based on the ACL stage.
-
         """
         pass
 
+    def send_and_check_mirror_packets(self,
+                                      setup,
+                                      mirror_session,
+                                      ptfadapter,
+                                      duthost,
+                                      mirror_packet,
+                                      src_port=None,
+                                      dest_ports=None,
+                                      expect_recv=True):
+        expected_mirror_packet = self._get_expected_mirror_packet(mirror_session,
+                                                                  setup,
+                                                                  duthost,
+                                                                  mirror_packet)
+
+        if not src_port:
+            src_port = self._get_random_src_port(setup)
+
+        if not dest_ports:
+            dest_ports = [self._get_monitor_port(setup, mirror_session, duthost)]
+
+        ptfadapter.dataplane.flush()
+        testutils.send(ptfadapter, src_port, mirror_packet)
+
+        if expect_recv:
+            _, received_packet = testutils.verify_packet_any_port(
+                ptfadapter,
+                expected_mirror_packet,
+                ports=dest_ports
+            )
+
+            logging.info("received: %s", packet.Ether(received_packet).summary())
+
+            inner_packet = self._extract_mirror_payload(received_packet, len(mirror_packet))
+            logging.info("inner_packet: %s", inner_packet.summary())
+            logging.info("expected_packet: %s", mirror_packet.summary())
+            pytest_assert(Mask(inner_packet).pkt_match(mirror_packet), "Mirror payload does not match received packet")
```

Comment:
is this check valid for egress mirror ? do we get modified or unmodified inner packet

Reply:
In this particular set of cases the inner packet should be the same, with the TTL decremented by 1 since we're just forwarding the packet. This assumption probably won't hold up for encapsulated packets (like VXLAN), so it will need to be re-visited when we add test cases that include tunneling.

Reply:
Actually, this does make me think the TTL decrement may be in the wrong place. Let me fix and re-test that. :)

Reply:
@dall even regular routed packet might be having different L2 Header

Reply:
You are right, I am adding that as well.

Reply:
There seem to be a couple other things that are impacted, I will try to narrow it down.

Reply:
Fixed and verified, thanks for the catch, @abdosi!